lib/rubocop/cop/rspec/no_stringified_instance_double_constant.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Do not use a string as `instance_double` constant.
+      #
+      # @example
+      #   # bad
+      #   instance_double('Foo', failure_message, stubs)
+      #
+      #   # good
+      #   instance_double(Foo, failure_message, stubs)
+      #
+      class NoStringifiedInstanceDoubleConstant < Base
+        extend AutoCorrector
+
+        MSG = 'Do not use a string as `instance_double` constant.'
+        RESTRICT_ON_SEND = %i[instance_double].freeze
+
+        # @!method stringified_instance_double_const?(node)
+        def_node_matcher :stringified_instance_double_const?, <<~PATTERN
+          (send nil? :instance_double $...)
```

Comment:
The node pattern can also do the `str_type?` check, _and_ return just the string node instead of an array:
```suggestion
          (send nil? :instance_double $str ...)
```

Reply:
Fixed and refactored the cop to match. Thanks for the improvement!